### PR TITLE
KAFKA-12749: Changelog topic config on suppressed KTable lost

### PR DESCRIPTION
**Pull request by vichu**
Refactored `logConfig` to be passed appropriately when using `shutDownWhenFull` or `emitEarlyWhenFull`. Removed the constructor that doesn't accept a `logConfig` parameter so you're forced to specify it explicitly, whether it's empty/unspecified or not.

Ticket: https://issues.apache.org/jira/browse/KAFKA-12749

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by vichu** [comment]
@ableegoldman Would appreciate it if you can take a look at this PR when you get a chance.

**Comment by ableegoldman** [comment]
cc any of @cadonna @vvcephei @lct45 @wcarlson5 @mjsax @guozhangwang to review this

File: streams/src/test/java/org/apache/kafka/streams/kstream/SuppressedTest.java [hunk 1]
```diff
@@ -46,13 +48,13 @@ public void bufferBuilderShouldBeConsistent() {
         assertThat(
             "keys alone should be set",
             maxRecords(2L),
-            is(new EagerBufferConfigImpl(2L, MAX_VALUE))
+            is(new EagerBufferConfigImpl(2L, MAX_VALUE, Collections.emptyMap()))
```
Comment (wcarlson5) [inline streams/src/test/java/org/apache/kafka/streams/kstream/SuppressedTest.java:51]: @vichu Can you add some tests that don't just use an empty map?

**Review by wcarlson5 — COMMENTED** [review]
Thanks for the PR!

The implementation looks pretty good so far, but can we add some tests to make sure that configs are no longer los  if either `emitEarlyWhenFull` or `shutDownWhenFull` is set after the logging config?

File: streams/src/test/java/org/apache/kafka/streams/kstream/SuppressedTest.java (hunk 1 above)
Comment (vichu) [reply streams/src/test/java/org/apache/kafka/streams/kstream/SuppressedTest.java:51]: Sure. 
Added a few tests to verify the `logConfig` map as well.

**Comment by wcarlson5** [comment]
Sorry, this got lost in my inbox. @vichu if you can rebase I think its good to go. do you agree @ableegoldman ?

File: streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/EagerBufferConfigImpl.java [hunk 2]
```diff
@@ -28,13 +28,7 @@
...
-    public EagerBufferConfigImpl(final long maxRecords, final long maxBytes) {
-        this.maxRecords = maxRecords;
-        this.maxBytes = maxBytes;
-        this.logConfig = Collections.emptyMap();
-    }
-
-    private EagerBufferConfigImpl(final long maxRecords,
+    public EagerBufferConfigImpl(final long maxRecords,
                                   final long maxBytes,
                                   final Map<String, String> logConfig) {
```
Comment (cadonna) [inline streams/src/main/java/org/apache/kafka/streams/kstream/internals/suppress/EagerBufferConfigImpl.java]: nit
```suggestion
    public EagerBufferConfigImpl(final long maxRecords,
                                 final long maxBytes,
                                 final Map<String, String> logConfig) {
```

**Review by cadonna — APPROVED** [review]
Thanks for the PR @vichu!

LGTM!

I found just one nit!

**Comment by cadonna** [comment]
@vichu I took the liberty to commit the fix to the nit directly. I hope that is fine with you.

**Comment by cadonna** [comment]
I will merge as soon as the checks are done.

**Comment by cadonna** [comment]
The 7th build was green on the same commit. I do not know why a 8th build was started.
https://ci-builds.apache.org/blue/organizations/jenkins/Kafka%2Fkafka-pr/detail/PR-10664/7/pipeline/15

**Pull request merged**